lib/auth/v2/getCanonicalizedAmzHeaders.js

```diff
@@ -0,0 +1,44 @@
+'use strict'; // eslint-disable-line strict
+
+function getCanonicalizedAmzHeaders(headers) {
+    /*
+    Iterate through headers and pull any headers that are x-amz headers.
+    Need to include 'x-amz-date' here even though AWS docs
+    ambiguous on this.
+    */
+    const amzHeaders = Object.keys(headers)
+        .filter(val => val.substr(0, 6) === 'x-amz-')
+        .map(val => [val.trim(), headers[val].trim()]);
+    /*
+    AWS docs state that duplicate headers should be combined
+    in the same header with values concatenated with
+    a comma separation.
+    Node combines duplicate headers and concatenates the values
+    with a comma AND SPACE separation.
+    Could replace all occurrences of ', ' with ',' but this
+    would remove spaces that might be desired
+    (for instance, in date header).
+    Opted to proceed without this parsing since it does not appear
+    that the AWS clients use duplicate headers.
+    */
+
+    // If there are no amz headers, just return an empty string
+    if (amzHeaders.length === 0) {
+        return '';
+    }
+
+
+    // Sort the amz headers by key (first item in tuple)
+    amzHeaders.sort((a, b) => {
```

Comment:
Same as my previous comment, even if an enhancement was possible that's not the purpose of this PR ;)